File: Python/palindrome.py
def is_palindrome_number(number):
    num = abs(number)  # Take the absolute value of the input to handle -ve numbers
    palindrome_sum = 0

    while num > 0:
        digit = num % 10
        palindrome_sum = palindrome_sum * 10 + digit
        num //= 10
    # Compare with absolute value of the original number
    # returns True if palindrome_sum == abs(number) else False
    return palindrome_sum == abs(number)  

# User Input
number = int(input("Enter a number: "))

if is_palindrome_number(number):
    print(number, "is a palindrome number")
else:
    print(number, "is not a palindrome number")

File: c/Palindrome.c
#include <stdio.h>

int main() {
    int num, reversedNum = 0, originalNum, remainder;

    // Input the number
    printf("Enter a number: ");
    scanf("%d", &num);

    originalNum = num; // Store the original number

    // Reverse the number
    while (num != 0) {
        remainder = num % 10;      // Get the last digit
        reversedNum = reversedNum * 10 + remainder; // Build the reversed number
        num = num / 10;           // Remove the last digit
    }

    // Check if the original number is equal to the reversed number
    if (originalNum == reversedNum) {
        printf("%d is a palindrome.\n", originalNum);
    } else {
        printf("%d is not a palindrome.\n", originalNum);
    }

    return 0;
}
